libs/platform/src/lib/components/form/multi-combobox/multi-combobox/multi-combobox.component.ts

```diff
@@ -0,0 +1,346 @@
+        }
+    }
+
+    /** @hidden
+     * Method to set input text as item label.
+     */
+    setInputTextFromOptionItem(item: OptionItem): void {
+        if (this.mobile) {
+            this.inputText = item.label;
+            this.cd.detectChanges();
+
+            return;
+        }
+
+        this.inputText = item.label;
+        this.showList(false);
+    }
+
+    /** @hidden */
+    onInputKeyupHandler(): void {
+        if (this.inputText && (!this._suggestions || !this._suggestions.length)) {
+            this.inputText = this.inputText.slice(0, -1);
+
+            this.isSearchInvalid = true;
+            this.state = 'error';
+
+            this.searchTermChanged();
+        } else {
+            this.isSearchInvalid = false;
+            this.state = 'default';
+        }
+    }
+
+    /** @hidden */
+    onItemKeyDownHandler(event: KeyboardEvent, item: SelectableOptionItem): void {
+        // unselect all
+        if ((event.ctrlKey || event.metaKey) && event.shiftKey && KeyUtil.isKeyCode(event, A)) {
+            event.preventDefault();
+            this.handleSelectAllItems(false);
+            return;
+        }
+
+        // select all
+        if ((event.ctrlKey || event.metaKey) && KeyUtil.isKeyCode(event, A)) {
+            event.preventDefault();
+            this.handleSelectAllItems(true);
+            return;
+        }
+
+        if (!KeyUtil.isKeyCode(event, ENTER) && !KeyUtil.isKeyCode(event, SPACE)) {
+            return;
+        }
+
+        if (KeyUtil.isKeyCode(event, ENTER) && !this.mobile) {
+            this.close();
+
+            return;
+        }
+    }
+
+    /** @hidden Handle dialog dismissing, closes popover and sets backup data. */
+    dialogDismiss(backup: SelectableOptionItem[]): void {
+        this._selected = [...backup];
+        this.inputText = '';
+        this.showList(false);
+        this.selectedShown$.next(false);
+    }
+
+    /** @hidden Handle dialog approval, closes popover and propagates data changes. */
+    dialogApprove(): void {
+        this._propagateChange(true);
+        this.inputText = '';
+        this.showList(false);
+    }
+
+    /** @hidden */
+    private _getTokenIndexByLabelOrValue(item: SelectableOptionItem): number {
+        return this._selected.findIndex(token => (token.label === item.label) || (token.value === item.value));
+    }
+
+    /** @hidden */
+    private _setSelectedItems(): void {
+        if (!this.selectedItems?.length) {
+            return;
+        }
+
+        for (let i = 0; i <= this.selectedItems.length; i++) {
+            const selectedItem = this.selectedItems[i];
+            const idx = this._flatSuggestions.findIndex(item => (item.label === selectedItem) || (item.value === selectedItem));
+            if (idx !== -1) {
+                this._selected.push(this._flatSuggestions[idx]);
+                this._flatSuggestions[idx].selected = true;
+            }
+        }
+
+
+        this.cd.detectChanges();
```

Comment:
same

Reply:
@Betrozov here `markForCheck` not checks changes till the change detection will be triggered _(so initially checked items won't be checked until something will be clicked/focused/etc.)_